app/processImage.js

```diff
@@ -48,26 +47,68 @@ function getCropFunction(cropType) {
 
 
 function applySmartCrop(image, cropSize, callback) {
-    smartcrop.crop(image, cropSize).then(function(result) {
-        var crop = result.topCrop;
-        sharp(image)
-            .extract({ width: crop.width, height: crop.height, left: crop.x, top: crop.y })
-            .resize(cropSize.width, cropSize.height)
-            .toBuffer(callback)
-    }, callback);
-}
+  const img = sharp(image);
 
-function applyCrop(image, cropSize, callback) {
-    sharp(image)
+  if (!(cropSize.height && cropSize.width)){
+    var asp_ratio;
+
+    img
+    .metadata()
+    .then(function(metadata) {
+      asp_ratio = metadata.width/metadata.height;
```

Comment:
The aspect ratio is has been calculated in the callback. There is no guarantee that this code will be executed before https://github.com/harshal27/image-resize-lambda/blob/0bccecf01fa37dfc238fa75800ca8c8063cf1b89/app/processImage.js#L64

Also this seems to be cleaning of cropSize params I would suggest to move this in async.waterfall in  processImage as a step 2 to make it separate from actual processing function. . Let me know if you have some clarification or have some other thoughts :)

Reply:
totally with u :)